### ISSUE TITLE
[Medium] setRevenueWallet has no timelock — admin compromise instantly redirects in-flight funds

### ISSUE THREAD

**Issue by clawdbotatg** [clawdbotatg/leftclaw-service-job-80#3]
## Severity
**Medium**

## Location
- `packages/foundry/contracts/PackShop.sol:108-113`
- `packages/foundry/contracts/TraitShop.sol:112-117`

```solidity
function setRevenueWallet(address newWallet) external onlyOwner {
    if (newWallet == address(0)) revert ZeroAddress();
    address prev = revenueWallet;
    revenueWallet = newWallet;
    emit RevenueWalletUpdated(prev, newWallet);
}
```

## Description
A compromised admin (or a key-handoff mistake) can frontrun pending pack/trait purchases by setting `revenueWallet` to an attacker address. The next purchase forwards `priceWei` (or `priceUsdc`) to the attacker. USDC transfers are irreversible.

This is the textbook admin-key-compromise finding for every admin-rotatable receiver. It's worth flagging Medium because the LeftClaw deploy doc explicitly says admin holds DEFAULT_ADMIN_ROLE on a fresh EOA at deploy and is expected to rotate to a Safe later — until that rotation, EOA loss = funds redirect.

## Recommended fix
Implement a 2-step revenue wallet flow with a short delay (12-24h):

```solidity
address public pendingRevenueWallet;
uint256 public revenueWalletEffectiveAt;
uint256 public constant REVENUE_WALLET_TIMELOCK = 24 hours;

event RevenueWalletProposed(address indexed proposed, uint256 effectiveAt);

function proposeRevenueWallet(address newWallet) external onlyOwner {
    if (newWallet == address(0)) revert ZeroAddress();
    pendingRevenueWallet = newWallet;
    revenueWalletEffectiveAt = block.timestamp + REVENUE_WALLET_TIMELOCK;
    emit RevenueWalletProposed(newWallet, revenueWalletEffectiveAt);
}

function acceptRevenueWallet() external onlyOwner {
    if (pendingRevenueWallet == address(0)) revert ZeroAddress();
    if (block.timestamp < revenueWalletEffectiveAt) revert TimelockNotElapsed();
    address prev = revenueWallet;
    revenueWallet = pendingRevenueWallet;
    pendingRevenueWallet = address(0);
    emit RevenueWalletUpdated(prev, revenueWallet);
}
```

## Specialist
Access control

**Comment by clawdbotatg** [comment]
Fixed in ffbf9c8 (Stage 6).
